katalog/istio/README.md

```diff
@@ -17,169 +34,175 @@ Source: [https://istio.io/docs](https://istio.io/docs/concepts/what-is-istio/#wh
-  use destination rules to configure what happens to traffic for that destination.
-  - **Gateways:** You use a gateway to manage inbound and outbound traffic for your mesh, letting you specify which
-  traffic you want to enter or leave the mesh.
-  **IMPORTANT NOTE** *(Only ingresss gateway is deployed with the current version)*
-  - **Service entries:** You use a service entry to add an entry to the service registry that Istio maintains
-  internally
-  - **Sidecars:** You can use a sidecar configuration to do the following:
-    - Fine-tune the set of ports and protocols that an Envoy proxy accepts.
-    - Limit the set of services that the Envoy proxy can reach.
-- ***Network resilience and testing:*** Istio provides opt-in failure recovery and fault injection features that you
-can configure dynamically at runtime.
-  - **Timeouts:** A timeout is the amount of time that an Envoy proxy should wait for replies from a given service,
-  ensuring that services don’t hang around waiting for replies indefinitely and that calls succeed or fail within a
-  predictable timeframe.
-  - **Retries:** A retry setting specifies the maximum number of times an Envoy proxy attempts to connect to a service
-  if the initial call fails.
-  - **Circuit breakers:** In a circuit breaker, you set limits for calls to individual hosts within a service, such
-  as the number of concurrent connections or how many times calls to this host have failed.
-  - **Fault injection:** Fault injection is a testing method that introduces errors into a system to ensure that it
-  can withstand and recover from error conditions.
-- ***Policies:*** Istio lets you configure custom policies for your application to enforce rules at runtime such as:
-  - Rate limiting to dynamically limit the traffic to a service
-  - Denials, whitelists, and blacklists, to restrict access to services
-  - Header rewrites and redirects
-- ***Observability:*** Istio generates detailed telemetry for all service communications within a mesh. This telemetry
-provides observability of service behavior, empowering operators to troubleshoot, maintain, and optimize their
-applications – without imposing any additional burdens on service developers.
-
-## Image repository and tag
-
-All istio container images are currently available at dockerhub: [docker.io/istio](https://hub.docker.com/u/istio)
-
-* istio container images: `docker.io/istio/*`
-
-Includes:
-
-- docker.io/istio/proxyv2
-- docker.io/istio/mixer
-- docker.io/istio/pilot
-
-* addons container images:
-
-- Kiali: quay.io/kiali/kiali:v1.9
-
-## Deployment
-
-You can deploy istio by running following command in the root of the project:
-
-```shell
-$ kustomize build | kubectl apply -f -
+In order to install the Istio package, it's required to have installed the [init](./init/README.md) package.
+This is the one involved in the Istio CRDs creation.
```

Comment:
Shouldn't the `init` package be a "module" of the Istio package? why is it a different package if it creates the CRDs needed by Istio?

Reply:
While developing i were thinking in the easiest way to avoid race conditions. Istio provides jobs to create CRDs in a specific order. This is the principal reason i put the "init" package separated from the "minimal" istio installation.

If we put this jobs along with the minimal istio installation we will end in some errors due to race conditions.

If you found a better way to avoid this kind of race conditions, let us know.

Reply:
I agree with @angelbarrera92 about the current implementation, let's keep `init` as a separate kustomize base.